src/AppBundle/Controller/RegistrationController.php

```diff
@@ -0,0 +1,196 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: arnoldas
+ * Date: 16.10.30
+ * Time: 19.33
+ */
+
+namespace AppBundle\Controller;
+
+
+
+use AppBundle\Form\ResetPasswordType;
+use AppBundle\Form\SendEmailPswResetType;
+use FOS\UserBundle\FOSUserBundle;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\Config\Definition\Exception\Exception;
+use Symfony\Component\HttpFoundation\Request;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use AppBundle\Entity\Post;
+use AppBundle\Entity\User;
+use AppBundle\Form\PostType;
+use AppBundle\Form\LoginType;
+use AppBundle\Form\UserType;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use FOS\UserBundle\Form\Type\RegistrationFormType;
+class RegistrationController extends Controller
+{
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    /**
+     * @Route("/registration", name="app.registration")
+     */
+    public function createUser(Request $request)
+    {
+        $user = new User();
+
+        $form = $this->createForm(UserType::class, $user);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            /** @var User $user */
+            $user = $form->getData();
+            $user->setEnabled(false);
+            // random hash used for confirmation token
+            $random_hash = md5(uniqid(rand(), true));
+            $user->setConfirmationToken($random_hash);
+            $user->addRole('ROLE_USER');
+            $this->sendAction($user,$random_hash);
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($user);
+            $em->flush();
+            return $this->redirectToRoute('app.successReg', ['id' => $user->getId()]);
+        }
+
+        return $this->render('AppBundle:Post:create.html.twig',
+            [
+                'form' => $form->createView()
+            ]
+        );
+    }
+    /**
+     * @Route("/successReg/{id}", name="app.successReg")
+     */
+    public function showSuccessRegistration($id)
+    {
+        $exampleService = $this->get('app.example');
+        return $this->render('AppBundle:Post:successReg.html.twig',
+            [
+
+                'postas' =>  $exampleService->getUserById($id),
+
+            ]
+        );
+    }
+
+
+    public function sendAction(User $user, $confirmationToken)
```

Comment:
File structure is random, would be much nicer to keep it clean, single new line before each method, no extended empty spaces like array above, etc